src/status.ts

```diff
@@ -1,259 +1,454 @@
 
-  private _testResults: BasicTestResults|null = null;
-  public get testResults(): BasicTestResults|null { return this._testResults; }
-  public set testResults(v: BasicTestResults|null) {
-    this._testResults = v;
+  private _enabled: boolean = false;
+  private _results: BasicTestResults|null = null;
+  private _color: string = '';
 
+  set enabled(v: boolean) {
+    this._enabled = v;
+    this.update();
+  }
+  set results(v: BasicTestResults|null) {
+    this._results = v;
     if (!v) {
-      this._testButton.text = localize('run.ctest', 'Run CTest');
-      this._testButton.color = '';
-      return;
+      this._color = '';
+    } else {
+      this._color = v.passing === v.total ? 'lightgreen' : 'yellow';
     }
+    this.update();
+  }
 
-    const passing = v.passing;
-    const total = v.total;
-    const good = passing == total;
-    const icon = good ? 'check' : 'x';
-    let testPassingText: string;
-    if (total == 1) {
-      testPassingText = localize('test.passing', '{0}/{1} test passing', passing, total);
+  update(): void {
+    if (this._results) {
+      const {passing, total} = this._results;
+      this.icon = passing == total ? 'check' : 'x';
+    } else {
+      this.icon = 'beaker';
+    }
+    if (this.config.statusbar.advanced?.ctest?.color === true) {
+      this.button.color = this._color;
     } else {
-      testPassingText = localize('tests.passing', '{0}/{1} tests passing', passing, total);
+      this.button.color = '';
+    }
+    super.update();
+  }
+
+  protected isVisible(): boolean { return this._enabled; }
+
+  protected getTextNormal(): string {
+    if (!this._results) {
+      this.button.color = '';
+      return localize('run.ctest', 'Run CTest');
+    }
+    const {passing, total} = this._results;
+    if (total == 1) {
+      return localize('test.passing', '{0}/{1} test passing', passing, total);
     }
-    this._testButton.text = `$(${icon}) ${testPassingText}`;
-    this._testButton.color = good ? 'lightgreen' : 'yellow';
+    return localize('tests.passing', '{0}/{1} tests passing', passing, total);
   }
 
-  /** Reloads the content of the build button */
-  private _reloadBuildButton() {
-    this._buildButton.text = ``;
-    this._buildButton.text = this._isBusy ? `$(x) ${localize('stop', 'Stop')}` : `$(gear) ${localize('build', 'Build')}:`;
-    this._buildButton.command = this._isBusy ? 'cmake.stop' : 'cmake.build';
-    if (this._isBusy) {
-      this._buildButton.show();
+  protected getTextShort(): string {
+    if (!this._results) {
+      return '-';
     }
+    const {passing, total} = this._results;
+    return `${passing}/${total}`;
   }
+}
+
+class BuildButton extends Button {
+  private static readonly _build = localize('build', 'Build');
+  private static readonly _stop = localize('stop', 'Stop');
+
+  settingsName = 'build';
+  command = 'cmake.build';
+  tooltip = localize('build.tooltip', 'Build the selected target');
```

Comment:
While testing this, I thought the tooltip looked strange with all of that whitespace on the second line.  I would prefer that the [target] be on the same line as the description.

I'll add a comment to each of the classes where I think some changes should be made to the tooltips.

Reply:
removed the newline in Debug, Launch and Build.